ChatRPG/ChatRPG/Data/FileUtility.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System.Text;
+
+namespace ChatRPG.Data;
+
+public record MessagePair(string PlayerMessage, string AssistantMessage);
+
+public class FileUtility
+{
+    private readonly string _path;
+    private readonly string _saveDir;
+
+    private readonly string _filenamePrefix = "conversation";
+
+    // Define "special" keywords for determining the author of a message.
+    private readonly string _playerKeyword = "#<Player>: ";
+    private readonly string _gameKeyword = "#<Game>: ";
+
+    public FileUtility(string saveDir = "Saves/")
+    {
+        _saveDir = Path.Join(AppDomain.CurrentDomain.BaseDirectory, saveDir);
+        _path = SetPath(DateTime.Now);
+    }
+
+    public async Task UpdateSaveFileAsync(MessagePair messages)
+    {
+        // According to .NET docs, you do not need to check if directory exists first
+        Directory.CreateDirectory(_saveDir);
+
+        await using FileStream fs =
+            new FileStream(_path, FileMode.Append, FileAccess.Write, FileShare.None, bufferSize: 4096,
+                useAsync: true);
+
+        byte[] encodedPlayerMessage = Encoding.Unicode.GetBytes(PrepareMessageForSave(messages.PlayerMessage, true));
+        await fs.WriteAsync(encodedPlayerMessage, 0, encodedPlayerMessage.Length);
+
+        byte[] encodedAssistantMessage = Encoding.Unicode.GetBytes(PrepareMessageForSave(messages.AssistantMessage));
+        await fs.WriteAsync(encodedAssistantMessage, 0, encodedAssistantMessage.Length);
```

Comment:
Det er mange async keywords på kode som bliver evaluated meget synchronously (pga. awaits!) - Kan man kombinere de to beskeder og skrive á én omgang, og så ikke await det, men låse filen imens? Så skulle det være thread safe, og så behøver systemet ikke at vente på at det er done

Reply:
Vi snakkede om det fysisk. Jeg går ud fra din comment blev resolved 😗